src/polyfill/bigint64array.js

```diff
@@ -0,0 +1,94 @@
+// BigInt ==> globalThis so we can use globalThis.
+var BigInt64Array;
+var BigUint64Array;
+if (typeof globalThis.BigInt64Array !== "undefined") {
+    BigInt64Array = globalThis.BigInt64Array;
```

Comment:
Every browser that supports BigInt supports globalThis so I think it's okay.

Reply:
This way avoids adding the polyfill to the global scope it only adds it inside of Emscripten which seems preferable?

Reply:
It seems cleaner to just add it to the global scope. That is generally what polyfills do.. and that is why we have the `-sNO_POLYFILLS` option if users don't like it.

Also, isn't that `var BigInt64Array;` here at global scope anyway?

Reply:
At least if you use `-sMODULARIZE` it isn't. If you prefer I will change it as you suggested.

Reply:
Okay I updated it as you suggested.

Reply:
I'm not sure what the best practice is in the JS world for this, but I'm worried about writing to `globalThis`. Yes, `var foo` and `foo = ` also write to the global scope, but at least the user can do
```js
(function() {
  var foo;
  // all the rest of the code
})()
```
That would avoid those "global" operations from actually being global. But `globalThis` would definitely write to the global scope, unless the user also protected it in a similar way. But then the operation is on all of `globalThis` at once, and not selective.

And in particular, the situation with `MODULARIZE` that was mentioned is closely related to that point. That mode can be helpful to avoid polluting the global namespace at all, which writing to `globalThis` would change.

But as I said, I'm not sure of JS conventions these days. Perhaps we should investigate those if we disagree here.

Reply:
I think it's better with the var approach but it isn't a very big difference.

Reply:
> I'm not sure what the best practice is in the JS world for this, but I'm worried about writing to `globalThis`. Yes, `var foo` and `foo = ` also write to the global scope, but at least the user can do
> 
> ```js
> (function() {
>   var foo;
>   // all the rest of the code
> })()
> ```
> 
> That would avoid those "global" operations from actually being global. But `globalThis` would definitely write to the global scope, unless the user also protected it in a similar way. But then the operation is on all of `globalThis` at once, and not selective.
> 
> And in particular, the situation with `MODULARIZE` that was mentioned is closely related to that point. That mode can be helpful to avoid polluting the global namespace at all, which writing to `globalThis` would change.
> 
> But as I said, I'm not sure of JS conventions these days. Perhaps we should investigate those if we disagree here.

The pollyfills are allowed to modify and add to the globalThis.. this is kind of point of them.   If folks don't want their runtime polyfilled we have the `-sNO_POLYFILL` option.   

For example, see the `Object.assign` polyfill.. it injects `Object.assign` onto `Object` which lives in the global namspace.

Reply:
I guess it doesn't really matter since most folks should not be using the polyfill anyway.

Reply:
I'm +1 to merge it as is.

Reply:
@sbc100 

> For example, see the Object.assign polyfill.. it injects Object.assign onto Object which lives in the global namspace.

Yes, but again, the user could do `var Object = ` in their scope, to "sandbox" that polyfill. That's easy, and then it wouldn't write to the global namespace. Writing to `globalThis` OTOH is harder for the user to work around.

Anyhow, if you don't find that convincing, I won't push the matter any more. I agree most users should not be using the polyfill anyhow, and also they can set `POLYFILL=false` to avoid it entirely.

Reply:
I'm ok landing this change, but it might be good to have at least one test the confirms the polyfill is working.

Reply:
Agreed. I will add some tests soon.